cmd/kubelet/app/server.go

```diff
@@ -1261,16 +1263,18 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	rl := make(v1.ResourceList)
 	for k, v := range m {
 		switch v1.ResourceName(k) {
-		// CPU, memory, local storage, and PID resources are supported.
-		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs:
-			q, err := resource.ParseQuantity(v)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse quantity %q for %q resource: %w", v, k, err)
-			}
-			if q.Sign() == -1 {
-				return nil, fmt.Errorf("resource quantity for %q cannot be negative: %v", k, v)
+		// CPU, memory, local storage, swap, and PID resources are supported.
+		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs, nodev1.ResourceSwap:
+			if v1.ResourceName(k) != nodev1.ResourceSwap || utilfeature.DefaultFeatureGate.Enabled(features.NodeSwap) {
+				q, err := resource.ParseQuantity(v)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse quantity %q for %q resource: %w", v, k, err)
+				}
+				if q.Sign() == -1 {
+					return nil, fmt.Errorf("resource quantity for %q cannot be negative: %v", k, v)
```

Comment:
nit: it's strange that previous formatter used `%q` for `v` while this one uses `%v`. Also semicolons are typically used for errors. I know it is not relevant to this PR, just caught my eye